js/src/views/Address/address.js

```diff
@@ -59,16 +70,82 @@ class Address extends Component {
     );
   }
 
-  renderActionbar () {
+  renderActionbar (contact) {
     const buttons = [
+      <Button
+        key='delete'
+        icon={ <ActionDelete /> }
+        label='delete address'
+        onClick={ this.toggleDeleteDialog } />
     ];
 
     return (
       <Actionbar
         title='Address Information'
-        buttons={ buttons } />
+        buttons={ contact.meta.deleted ? [] : buttons } />
+    );
+  }
+
+  renderDeleteConfirm () {
+    const { contacts } = this.props;
+    const { deleteDialog } = this.state;
+
+    if (!deleteDialog) {
+      return;
+    }
+
+    const { address } = this.props.params;
+    const contact = contacts[address];
+
+    return (
+      <ConfirmDialog
+        className={ styles.delete }
+        title='confirm removal'
+        visible
+        onNo={ this.toggleDeleteDialog }
+        onYes={ this.onDeleteConfirmed }>
+        <div className={ styles.hero }>
+          Are you sure you want to remove the following address from your addressbook?
+        </div>
+        <div className={ styles.info }>
+          <IdentityIcon
+            className={ styles.icon }
+            address={ address } />
+          <div className={ styles.nameinfo }>
+            <div className={ styles.header }>
+              { contact.name || 'Unnamed' }
+            </div>
+            <div className={ styles.address }>
+              { address }
+            </div>
+          </div>
+        </div>
+        <div className={ styles.description }>
+          { contact.meta.description || 'this is just some long description to make me feel happy about myself and to see what actually happens and just maybe we should throw some wrapping in here as well' }
+        </div>
+      </ConfirmDialog>
     );
   }
+
+  onDeleteConfirmed = () => {
+    const { api, router } = this.context;
+    const { contacts } = this.props;
+    const { address } = this.props.params;
+    const contact = (contacts || {})[address];
+
+    this.toggleDeleteDialog();
```

Comment:
In this place, it always *hides* the delete dialog, so I'd prefer `hideDeleteDialog` & `showDeleteDialog`, as they're more expressive. Reading `toggleDeleteDialog`, I had to stop and actually figure out if it makes sense.